socs/agents/vantagepro2/agent.py

```diff
@@ -120,12 +126,25 @@ def acq(self, session, params=None):
 
             while self.take_data:
                 pm.sleep()
+
+                if self.module is None:  # Try to re-initialize module if it fails
+                    if not self._initialize_module():
+                        time.sleep(30)  # wait 30 sec before trying to re-initialize
+                        continue
+
                 data = {
                     'timestamp': time.time(),
                     'block_name': 'weather',
                     'data': {}
                 }
-                data['data'] = self.module.weather_daq()
+                try:
+                    data['data'] = self.module.weather_daq()
+                except TimeoutError as e:
+                    self.log.warn(f"TimeoutError: {e}")
+                    self.module = None
+                    time.sleep(30)
```

Comment:
Maybe if the connection has dropped we just try again right away to initialize then let the above 30 second sleep take effect?